astropy/table/column.py

```diff
@@ -343,6 +343,79 @@ class ColumnInfo(BaseColumnInfo):
     """
     attrs_from_parent = BaseColumnInfo.attr_names
     _supports_indexing = True
+    # For structured columns, data is used to store a dict of columns.
+    # Store entries in that dict as name.key instead of name.data.key.
+    _represent_as_dict_primary_data = 'data'
+
+    def _represent_as_dict(self):
+        result = super()._represent_as_dict()
+        if names := self._parent.dtype.names:
+            # For a structured column, we use a SerializedColumns to
+            # store the pieces.
+            from .serialize import SerializedColumn
+
+            data = SerializedColumn()
+            # If this column has a StructuredUnit, we split it and store
+            # it on the corresponding part. Otherwise, we just store it
+            # as an attribute below.  All other attributes we remove from
+            # the parts, so that we do not store them multiple times.
+            # TODO: deal with (some of) this in Column.__getitem__?
+            # Alternatively: should we store info on the first part?
+            # TODO: special-case format somehow? Can we have good formats
+            # for structured columns?
+            unit = self.unit
+            if isinstance(unit, StructuredUnit) and len(unit) == len(names):
+                units = unit.values()
+                unit = None  # No need to store as an attribute as well.
+            else:
+                units = [None] * len(names)
+            for name, part_unit in zip(names, units):
+                part = self._parent[name]
+                part.unit = part_unit
+                part.description = None
+                part.meta = {}
+                part.format = None
+                data[name] = part
+
+            # Create the attributes required to reconstruct the column.
+            result['data'] = data
+            # Store the shape if needed to distinguish, e.g., a column
+            # with dtype '2f8,2i8' and just one element per row from a
+            # column with dtype with 'f8,i8' and 2 elements per row.
+            if shape := self._parent.shape[1:]:
+                result['shape'] = list(shape)
+            # Also store the standard info attributes since these are
+            # stored on the parent and can thus just be passed on as
+            # arguments.  TODO: factor out with essentially the same
+            # code in serialize._represent_mixin_as_column.
+            if unit is not None and unit != '':
+                result['unit'] = unit
+            if self.format is not None:
+                result['format'] = format
+            if self.description is not None:
+                result['description'] = self.description
+            if self.meta:
+                result['meta'] = self.meta
+
+        return result
+
+    def _construct_from_dict(self, map):
+        if not hasattr(map.get('data', None), 'items'):
```

Comment:
The `None` is automatic

```suggestion
        if not hasattr(map.get('data'), 'items'):
```

Reply:
Done